sdks/python/tests/testlib/assert_helpers.py

```diff
@@ -13,7 +13,23 @@ def prepare_difference_report(expected: Any, actual: Any) -> str:
         diff_report = deepdiff.DeepDiff(
             expected, actual, exclude_types=[mock.mock._ANY]
         ).pretty()
-        return diff_report
+
+        # Remove from report lines like that "X type changed from int to ANY_BUT_NONE"
+        # But keep the lines like "X type changed from NoneType to ANY_BUT_NONE"
+        # The rest of the lines remain.
+        diff_report_lines = diff_report.split("\n")
+        diff_report_cleaned_lines = [
+            diff_report_line
+            for diff_report_line in diff_report_lines
+            if (
+                "NoneType to AnyButNone" in diff_report_line
+                or "AnyButNone to NoneType" in diff_report_line
+                or "AnyButNone" not in diff_report_line
+            )
+        ]
+        diff_report_clean = "\n".join(diff_report_cleaned_lines)
```

Comment:
Not a big fan of altering reports, as context is lost over time in the future. But not a blocker.

Reply:
1. This is a self-made comparison reporting
2. This cleaning helps to keep only the things that are relevant. Our tests has a lot of ANY, ANY_BUT_NONE, ANY_DICT objects, this cleaning removes lines like "Value changed from 42 to ANY_BUT_NONE". I'll keep it.